packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -956,19 +956,71 @@ const alternateTypeKey = createStateSymbol("alternateType");
 export const $alternateType: AlternateTypeDecorator = (
   context: DecoratorContext,
   source: ModelProperty | Scalar,
-  alternate: Scalar,
+  alternate: Type,
   scope?: LanguageScopes,
 ) => {
-  if (source.kind === "ModelProperty" && source.type.kind !== "Scalar") {
+  let invalidAlternate = true;
+  let invalidSource = true;
+
+  // if alternate type is unknown, source type must be scalar
+  if (alternate.kind === "Intrinsic" && alternate.name === "unknown") {
+    invalidAlternate = false;
+    if (
+      (source.kind === "ModelProperty" && source.type.kind === "Scalar") ||
+      source.kind === "Scalar"
+    ) {
+      invalidSource = false;
+    }
+  }
+
+  // if alternate type is scalar array, source type must be scalar array
```

Comment:
how about array of array, array of map? we do not support that for now? i don't like add too much validation for this decorator and let tsp author to take the responsibility.

Reply:
> how about array of array, array of map? we do not support that for now? i don't like add too much validation for this decorator and let tsp author to take the responsibility.

array of array or array of map is not supported. @pshao25 , am I right here?